shell/browser/native_window_mac.mm

```diff
@@ -1403,15 +1403,35 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   original_frame_ = [window_ frame];
 }
 
-void NativeWindowMac::SetVibrancy(const std::string& type) {
+void NativeWindowMac::SetVibrancy(const std::string& type, bool animate) {
   NSVisualEffectView* vibrantView = [window_ vibrantView];
 
   if (type.empty()) {
     if (vibrantView == nil)
       return;
 
-    [vibrantView removeFromSuperview];
-    [window_ setVibrantView:nil];
+    vibrancy_type_ = type;
+
+    if (animate) {
+      __weak auto weak_delegate = window_delegate_.get();
+      [NSAnimationContext
+          runAnimationGroup:^(NSAnimationContext* context) {
+            context.duration = 0.3;
+            vibrantView.animator.alphaValue = 0.0;
+          }
+          completionHandler:^{
+            if (!weak_delegate)
+              return;
+
+            if (vibrancy_type_.empty()) {
+              [vibrantView removeFromSuperview];
```

Comment:
We need to guard against multiple calls of animated `setVibrancy`

Reply:
multiple calls shouldn't be a problem since if you animate in a vibrancy before the remove completion runs, it will reuse the vibrant view still there so the completion won't remove it since the type will be already set again